packages/core/src/decorators/oas2/remove-unused-components.ts

```diff
@@ -16,15 +16,44 @@ export const RemoveUnusedComponents: Oas2Decorator = () => {
     name: string
   ): void {
     components.set(location.absolutePointer, {
-      used: components.get(location.absolutePointer)?.used || false,
+      usedIn: components.get(location.absolutePointer)?.usedIn ?? [],
       componentType,
       name,
     });
   }
 
+  function removeUnusedComponents(root: Oas2Definition, removedPaths: string[]): number {
+    const removedLengthStart = removedPaths.length;
+
+    for (const [path, { usedIn, name, componentType }] of components) {
+      const used = usedIn?.some(
```

Comment:
```suggestion
      const used = usedIn.some(
```

Is the optional chaining still needed?

Reply:
It is not!